File: sdk/nodejs/examples/03_stream.ts
/**
 * Example 03 — continuously capture 10 frames, then auto-stop.
 *
 * Usage: H1_PORT=/dev/cu.usbserial-XXX npx tsx examples/03_stream.ts
 */

import { Device, type SpectrumFrame } from '../src/index.js';

const port = process.env.H1_PORT;
if (!port) {
  process.stderr.write('Set H1_PORT to the serial port path before running.\n');
  process.exit(1);
}

const TARGET = 10;
const device = new Device(port);

try {
  let received = 0;
  const done = new Promise<void>((resolve, reject) => {
    device.on('frame', (frame: SpectrumFrame) => {
      received++;
      process.stdout.write(
        `[${received}/${TARGET}] CCT=${frame.photometric.CCT.toFixed(0)}K  ` +
          `lux=${frame.photometric.lux.toFixed(1)}\n`,
      );
      if (received >= TARGET) {
        device.stopStreaming().then(resolve, reject);
      }
    });
    device.on('error', reject);
  });

  await device.startStreaming(false);
  await done;
} finally {
  await device.close();
}

File: sdk/python/examples/03_stream.py
"""Example 3: stream 10 frames and print a one-line summary each.

Usage:
    python examples/03_stream.py /dev/tty.usbserial-XYZ
"""

from __future__ import annotations

import sys

from h1_sdk import Device


def main(port: str) -> None:
    with Device(port) as dev:
        for i, frame in enumerate(dev.stream(max_frames=10)):
            p = frame.photometric
            print(
                f"#{i:02d} status={frame.exposure_status.name:6s} "
                f"exposure={frame.exposure_time_us:>7d}us "
                f"CCT={p.CCT:7.1f}K Ra={p.Ra:5.2f} lux={p.lux:8.2f}"
            )


if __name__ == "__main__":
    if len(sys.argv) != 2:
        print(f"usage: {sys.argv[0]} <serial-port>", file=sys.stderr)
        sys.exit(2)
    main(sys.argv[1])
